src/core/u-compress.c

```diff
@@ -355,9 +406,10 @@ REBSER *Decompress(
 
     // Since the initialization succeeded, go ahead and make the output buffer
     //
-    REBSER *output = Make_Binary(buf_size);
+    REBSER *output = Make_Binary(sizeof(REBSER*) + buf_size);
+    *cast(REBSER**, BIN_HEAD(output)) = output; // self-referential prefix
     strm.avail_out = buf_size;
-    strm.next_out = BIN_HEAD(output);
+    strm.next_out = BIN_HEAD(output) + sizeof(REBSER*);
 
```

Comment:
It would be better if this level of details could be hidden in a function close to Rebol_Malloc. Maybe something like REBSER *Rebol_Malloc_Setup/Prepare(size_t) and Rebol_Malloc_From(REBSER *)